src/haversine.jl

```diff
@@ -12,7 +12,7 @@ end
 
 const VecOrLengthTwoTuple{T} = Union{AbstractVector{T}, NTuple{2, T}}
 
-function evaluate(dist::Haversine, x::VecOrLengthTwoTuple, y::VecOrLengthTwoTuple) 
+function evaluate(dist::Haversine, x::VecOrLengthTwoTuple{T}, y::VecOrLengthTwoTuple{T}) where {T}
```

Comment:
Do you really need to restrict both vectors to be of the same type? You could just call `one(a)` below, which would be more correct anyway.

Reply:
Very good point, I will update the PR accordingly.

Reply:
Thanks. Somebody else will have to review the math as I have no idea what this is doing...

Reply:
Sometimes due to floating point errors this variable "a" becomes something like "a = 1.000000001". This change is only discarding the noise.